scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -338,49 +323,28 @@ def matmat_XH_X(x):
     elif solver == 'arpack' or solver is None:
         if v0 is None and not rs_was_None:
             v0 = random_state.uniform(size=(min(A.shape),))
-        eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
+        _, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
                                 ncv=ncv, which=which, v0=v0)
 
-    # Gramian matrices have real non-negative eigenvalues.
-    eigvals = np.maximum(eigvals.real, 0)
-
-    # Use the sophisticated detection of small eigenvalues from pinvh.
-    t = eigvec.dtype.char.lower()
-    factor = {'f': 1E3, 'd': 1E6}
-    cond = factor[t] * np.finfo(t).eps
-    cutoff = cond * np.max(eigvals)
-
-    # Get a mask indicating which eigenpairs are not degenerately tiny,
-    # and create the re-ordered array of thresholded singular values.
-    above_cutoff = (eigvals > cutoff)
-    nlarge = above_cutoff.sum()
-    nsmall = k - nlarge
-    slarge = np.sqrt(eigvals[above_cutoff])
-    s = np.zeros_like(eigvals)
-    s[:nlarge] = slarge
+    eigvec, _ = np.linalg.qr(eigvec)
+    u = X_matmat(eigvec)
     if not return_singular_vectors:
-        return np.sort(s)
-
-    if n > m:
-        vlarge = eigvec[:, above_cutoff]
-        ularge = (X_matmat(vlarge) / slarge
-                  if return_singular_vectors != 'vh' else None)
-        vhlarge = _herm(vlarge)
+        s = svd(u, compute_uv=False, overwrite_a=True)
+        return s[::-1]
+
+    # compute the left singular vectors of X and update the right ones
+    # accordingly
+    u, s, vh = svd(u, full_matrices=False, overwrite_a=True)
```

Comment:
One QR and one SVD looks unnecessarily expensive for the returns we are getting here in my opinion. All we wanted to have is a bit more stability but we have gone into two intensive decompositions already. In my computer svds is 10% slowed down with not so much precision increase.

Reply:
The QR is somewhat optional, needed in cases where the eigensolvers output eigenvectors that are not orthogonal enough leading to not quite orthogonal singular vectors. If removed, unit tests that check orthogonality of the final singular vectors would fail. 

The current master unit tests simply do not even check orthogonality of the singular vectors, probably because such tests would just fail in the current master without a possibility of a fix.